keps/sig-auth/20190730-oidc-discovery.md

```diff
@@ -195,10 +243,70 @@ configuration:
 ```
 
 The API server would treat these as `nonResourceURLs`, and restrict access
-appropriately. We will consider expanding `system:public-info-viewer` RBAC
-ClusterRole to grant access to the new paths; some other roles may have
-permission already via a pattern match on `nonResourceURLs` (e.g.
-`cluster-admin`).
+appropriately. We will provide a default RBAC `ClusterRole` called
+`service-account-issuer-discovery` that provides `GET` access to these
+`nonResourceURLs`, but will *not* provide a default `ClusterRoleBinding`. This
+leaves the decision of who is allowed to access these endpoints up to cluster
+admins. A default binding requires further discussion, including ongoing efforts
+to harden the unauthenticated API surface area.
+
+### Risks and Mitigations
+
+- Security is being reviewed by @mikedanese and @liggitt.
+- This feature exposes public key information that is derived from sensitive
+  private keys. It is important that code reviewers pay careful attention
+  to the construction of the keysets so that sensitive private keys are not
+  exposed.
+- This proposal allows Kubernetes identities to be federated into other systems,
+  and creates a dependency on the Kubernetes API server to verify these identities.
+  This means that if the API server is down, and verification keys are not
+  cached by relying parties or an intermediary, workloads may fail to
+  authenticate with their dependencies. This can be mitigated by running a high
+  availability configuration, or by caching discovery docs and keysets in a
+  reliable intermediate location.
+
+## Design Details
+
+### Test Plan
+
+This feature includes unit, integration, and E2E tests:
+- Unit tests in `pkg/serviceaccount` to test that the server code constructs
+  OIDC responses in the correct format.
+- Integration tests in `test/integration/auth/svcacct_test.go` to attempt a
+  token verification based on the OIDC flow against an API server, and also
+  to verify the expected headers on the responses.
+- An E2E test that exercises the full stack by mounting a new-style Kubernetes
```

Comment:
An e2e test that confirms a pod running in cluster A can validate a KSA from cluster B would also be interesting.  Maybe too much of a hassle to set up, but I think it is a valid scenario to consider.

Reply:
Yeah that could be interesting. I imagine if relying parties outside the cluster can validate it, then pods in some other cluster should also be able to as well (as long as they have a way to make a request outside their host cluster).